java/connectors/semantickernel-connectors-memory-redis/src/main/java/com/microsoft/semantickernel/connectors/memory/redis/RedisMemoryRecord.java

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.connectors.memory.redis;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.semantickernel.ai.embeddings.Embedding;
+import com.microsoft.semantickernel.memory.DataEntryBase;
+import com.microsoft.semantickernel.memory.MemoryRecord;
+import redis.clients.jedis.search.Document;
+
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/** Represents an entry in the Semantic Kernel Memory Table. */
+public class RedisMemoryRecord extends JsonMemoryRecord {
+
+    public RedisMemoryRecord(String id, String text, String description, String additionalMetadata,
+                             List<Float> embedding, String externalSourceName, boolean isReference) {
+        super(id, text, description, additionalMetadata, embedding, externalSourceName, isReference);
+    }
+
+    /**
+     * Converts a RedisSearch Document to a HashMap.
+     *
+     * @param document A Redis Search Document
+     *
+     * @return A HashMap containing the iterator's entries
+     */
+    protected Map<String, Object> documentToMap(Document document) {
+        return StreamSupport.stream(document.getProperties().spliterator(), false)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    /**
+     * Gets the embedding information associated with the entry.
+     *
+     * @param map A hashmap with keys representing a record
+     * @param withEmbedding A boolean flag indicating that the embedding should be returned with the metadata
+     *
+     * @return A MemoryRecord object.
+     */
+    public static MemoryRecord mapToRecord(Map<String, String> map, boolean withEmbedding) {
+        Embedding embedding = Embedding.empty();
+        ZonedDateTime time = null;
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+//            if (map.get(TIME) != null) {
```

Comment:
Why is this commented out?

Reply:
because it's not plain that it's used, but it has been uncommented and will be used, if present